### PR TITLE
fix(tpg v5)!: rule.rate_limit_options.enforce_on_key has not default value

### PR DESCRIPTION
**Pull request by imrannayer**
In `4.X`, the default value for `rule.rate_limit_options.enforce_on_key` is `ALL`. In `5.X` this field no longer has a default value. If you need `All` you will need to set it explicitly. See [Rule 2](https://github.com/GoogleCloudPlatform/terraform-google-cloud-armor/blob/main/examples/security-policy-all/main.tf) in `examples/security-policy-all` folder for reference.

### CODE REVIEW

File: versions.tf [hunk 1]
```diff
@@ -19,11 +19,11 @@ terraform {
   required_providers {
     google = {
       source  = "hashicorp/google"
-      version = ">= 4.79.0, < 5.0"
+      version = ">= 4.79.0, < 6.0"
```
Comment (g-awmalik) [inline versions.tf:22]:
```suggestion
      version = ">= 4.79.0, < 6"
```

File: versions.tf [hunk 2]
```diff
@@ -19,11 +19,11 @@ terraform {
...
     }
     google-beta = {
       source  = "hashicorp/google-beta"
-      version = ">= 4.79.0, < 5.0"
+      version = ">= 4.79.0, < 6.0"
```
Comment (g-awmalik) [inline versions.tf:26]:
```suggestion
      version = ">= 4.79.0, < 6"
```

File: examples/security-policy-all/versions.tf
Comment (g-awmalik) [inline examples/security-policy-all/versions.tf:1]: Do we need versions.tf for examples?

File: versions.tf (hunk 1 above)
Comment (apeabody) [inline versions.tf:22]:
```suggestion
      version = ">= 4.79.0, < 6"
```

File: versions.tf (hunk 2 above)
Comment (apeabody) [inline versions.tf:26]:
```suggestion
      version = ">= 4.79.0, < 6"
```

File: examples/security-policy-all/versions.tf [hunk 3]
```diff
@@ -18,11 +18,11 @@ terraform {
   required_providers {
     google = {
       source  = "hashicorp/google"
-      version = "~> 4.0"
+      version = ">= 4.0"
```
Comment (apeabody) [inline examples/security-policy-all/versions.tf:21]: For `/examples` you could also just remove the version line entirely, but this is also just fine.

File: versions.tf (hunk 1 above)
Comment (g-awmalik) [reply versions.tf:22]: Using a full major version constraint will help widen the range in the future.

**Review by g-awmalik — COMMENTED** [review]
Couple of minor nits.

**Review by apeabody — COMMENTED** [review]
Thanks @imrannayer! Can you also lift to `6` in https://github.com/GoogleCloudPlatform/terraform-google-cloud-armor/blob/main/test/setup/versions.tf

File: versions.tf (hunk 1 above)
Comment (apeabody) [reply versions.tf:22]: dup with g-awmalik

File: versions.tf (hunk 2 above)
Comment (apeabody) [reply versions.tf:26]: dup with g-awmalik

File: examples/security-policy-all/versions.tf
Comment (imrannayer) [reply examples/security-policy-all/versions.tf:1]: Since we dont have any other provider except google, leme remove it.

**Comment by apeabody** [comment]
> @g-awmalik @apeabody This will be a breaking change as there is a behavior change for `rule.rate_limit_options.enforce_on_key` in 5.x.

Thanks @imrannayer - @g-awmalik any thoughts on releasing https://github.com/GoogleCloudPlatform/terraform-google-cloud-armor/pull/70 as a final v4 compatible before merging this?

**Comment by imrannayer** [comment]
> > @g-awmalik @apeabody This will be a breaking change as there is a behavior change for `rule.rate_limit_options.enforce_on_key` in 5.x.
> 
> Thanks @imrannayer - @g-awmalik any thoughts on releasing #70 as a final v4 compatible before merging this?

@apeabody #70 is already a breaking change. I think it will be better to release 5.X with it.

**Comment by apeabody** [comment]
> > > @g-awmalik @apeabody This will be a breaking change as there is a behavior change for `rule.rate_limit_options.enforce_on_key` in 5.x.
> > 
> > 
> > Thanks @imrannayer - @g-awmalik any thoughts on releasing #70 as a final v4 compatible before merging this?
> 
> @apeabody #70 is already a breaking change. I think it will be better to release 5.X with it.



> > > @g-awmalik @apeabody This will be a breaking change as there is a behavior change for `rule.rate_limit_options.enforce_on_key` in 5.x.
> > 
> > 
> > Thanks @imrannayer - @g-awmalik any thoughts on releasing #70 as a final v4 compatible before merging this?
> 
> @apeabody #70 is already a breaking change. I think it will be better to release 5.X with it.

Just to confirm, the breaking change `rule.rate_limit_options.enforce_on_key` will also raise the minimum TPG to `>= 5`?

**Comment by imrannayer** [comment]
@apeabody I am not changing min version to 5.X. Min is still 4.79. But in case anyone gets upgrade to 5.X will need to make sure they provide a value for rule.rate_limit_options.enforce_on_key.

**Comment by apeabody** [comment]
> @apeabody I am not changing min version to 5.X. Min is still 4.79. But in case anyone gets upgrade to 5.X will need to make sure they provide a value for rule.rate_limit_options.enforce_on_key.

Got it - Thanks!

**Comment by imrannayer** [comment]
@g-awmalik can u plz approve this PR

File: examples/simple-example/versions.tf
Comment (g-awmalik) [inline examples/simple-example/versions.tf:1]: If possible, I would suggest deleting this file.

**Review by g-awmalik — APPROVED** [review]

**Pull request merged**